docs/DeveloperGuide.md

```diff
@@ -1,34 +1,346 @@
+testing and development sections to help developers better understand the application.
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The application has been broken down into different segments so that users would find it easy to navigate from one
+segment to the next one. There are three segments within the application:
+* Subjects: This segment edits the list of all the subjects present within the application and takes care of relevant
+commands such as add subject, delete subject or find subject.
+* Subject: This segment is accessed through the `subject NAMEOFSUBJECT` command. The user would then look at the individual 
+subjects and then would be able to add either tasks or topics.
+* Topic : This segment is accessed through the `topic NAMEOFTOPIC` command. The user can add, delete or find flashcards
+in this segment.
+
+The code is broken down into different packages, which combine classes that perform a similar functionality. Here
+are the list of packages:
+* Card: In charge of classes related to the subject, topic, flashcard and quiz functionalities
+    - Quiz
+        * Quiz
+        * Result
+        * ResultList
+        * TopicQuiz
+        * SubjectQuiz
+    - Flashcard
+    - Subject
+    - SubjectList
+    - Topic
+    - TopicList
+    
+
+* Command: In charge of reading the command and calling the relevant methods. The package itself holds
+the following packages.
+    - Flashcard
+    - Subject
+    - Topic
+    - Task
+    
+* Exceptions: Deals with the various exceptions which are thrown in the application.
+    -flashcard
+    - storage
+    - subject
+    - task
+    - topic
+* Parser: Parses the commands and instantiates the relevant command classes.
+* Storage: Stores the data in an external folder
+* Task: Contains the classes for the different types of tasks.
+* Ui: In charge of interaction with the user.
+
+### Quiz package
+The `Storage` package holds the necessary classes for the quiz functionality of this application. The following are
+the classes in the package:an abstract`Quiz` class, a `Result` class, a `SubjectQuiz` class and a `TopicQuiz` class.
+
+#### Quiz class
+`Quiz` class is an abstract class in the `Quiz` package. It is holds the result of a quiz class and the list of flashcards
+from any subject or topic classes which the user calls the quiz for. Furthermore, it contains a checkAnswer() method the
+checks the answer that the user had given with the correct answer of the quiz. If the user enters the correct answer, the 
+existing score is incremented by one. Else, the contents of the flashcards and the incorrect answer provided by
+the user is transferred to the  `incorrectAnswers` list. Once the user finished the quiz, the application would print the 
+questions that the users did not answer correctly, along the the answer that was provided by the user. 
```

Comment:
would it be better to have a class diagram? I felt the explanation is a bit wordy.